<button type="button" class="btn btn-primary btn-lg btn-block"><h2><div align="center">TP 1 : Groupe B1 </div></h2></button>

<button type="button" class="btn btn-info btn-lg btn-block"><h4><div align="left"><i>Exercice 1 : Types et Opérations</i></div></h4></button>

<div class="alert alert-danger" role="alert">   
    
  * Déclarer une variable ayant la valeur 34 et le type `int` 
    
  * Déclarer une variable ayant la valeur 34 et le type `float` 
    
  * Déclarer une variable ayant la valeur 34 et le type `string` 
    
  * Déclarer une variable ayant la valeur 34 et le type `list` 
    
   Vous pouvez vous aider de la fonction `type()` 
</div>

In [1]:
a = 
b =
c = 
d =


SyntaxError: invalid syntax (3231703949.py, line 1)

<div class="alert alert-danger" role="alert">
  Corrigez le code suivant :
</div>

In [2]:
a = input('saisissez un entier : ')
b = input('saisissez un 2e entier :')

a+=1
b-=1
inferieur == a<=b

print('a+1 =',a,' est plus grand ou égal à b-1 =',b,'? La réponse est :',not(inferieur))

          


KeyboardInterrupt: Interrupted by user

<div class="alert alert-warning" role="alert">
 Quel est le type de la variable superieur ?
</div>

### Réponse : 

<div class="alert alert-danger" role="alert">
  Expliquez la différence entre les deux afféctations suivante, et donnez leur type de chacune des variables :
</div>

In [ ]:
a = 1,618
b = 1.618

<div class="alert alert-danger" role="alert">
  On répertorie des gens avec leurs prénom, ville de naissance, date de Naissance et si ils sont déjà aller au ski (True ou False)
</div>

In [ ]:
liste = ['Eric','Ajaccio',[17,'Août',1985],True]

<div class="alert alert-warning" role="alert">
  Changer la ville de Naissance d'Eric par Lyon
</div>

<div class="alert alert-warning" role="alert">
 
Changer sa date d'anniversaire pour le 31 Août 1989

</div>

<div class="alert alert-warning" role="alert">
  Ajoutez à la fin de la liste l'animal préféré d'Eric, qui est la tortue
</div>

<div class="alert alert-warning" role="alert">
  Supprimer l'avant dernier élément de la liste
</div>

<button type="button" class="btn btn-info btn-lg btn-block"><h4><div align="left"><i>Exercice 3 : if else elif</i></div></h4></button>

<div class="alert alert-danger" role="alert">
  Tout les nombres entiers sont soit pair soit impair, c'est à dire qu'ils s'écrivent soit$ n=2 \times p$ soit $n=2\times p + 1$. Le code suivant donne ce test de parité :
</div>

In [ ]:
n=int(input('saissisez un entier'))

if n % 2 == 0 :
    print("n est paire et il s'écrit ",n,'= 2 *', n//2)
else :
    print("n est impaire et il s'écrit",n,'= 2 *',n//2,"+1")

<div class="alert alert-warning" role="alert">
De la même manière, écrivez un code pour vérifiez si ils s'écrivent soit $n=3\times p$ soit $n=3\times p+1$ soit $n=3\times p+2$
</div>

In [ ]:
n=int(input('saissisez un entier'))

<button type="button" class="btn btn-info btn-lg btn-block"><h4><div align="left"><i>Exercice 4 : Boucles while et for</i></div></h4></button>

Dans un premier temps on s'intéresse à la suite 
$$x_n=\frac{1}{2n+1}$$
Attention la suite part de $n=0$

<div class="alert alert-danger" role="alert">
Ecrivez un code qui print les 100 premiers termes de la suite

</div>

<div class="alert alert-warning" role="alert">
 Ecrivez un code qui donne print les éléments de la suite et qui s'arrête quand la suite est plus petite que 0.001
    ($x_n<0.001$)
</div>

In [ ]:
# Copiez votre code précédent et complétez le ici.

<div class="alert alert-warning" role="alert">
Ajoutez un compteur, qui compte le nombre d'itérations de la suite qu'il est nécéssaire de faire pour être plus petit que 0.001
</div>

In [ ]:
# Copiez votre code précédent et complétez le ici.

<div class="alert alert-danger" role="alert">
Ecrire un code qui print les tables de multiplications de 1 à 10.
    
Pour cela il faudrat utiliser les boucles imbriquées.

</div>

### Réponse : 

<div class="alert alert-danger" role="alert">
On chercher une approximation de $\pi$, pour cela on s'intéresse maintenant à la suite:
   $$x_n=\frac{-1^n}{2n+1}$$
    
Attention la suite part de $n=0$
    
    
Remarquez que $-1^n=1$ quand $n$ est paire et $-1^n=-1$ quand $n$ est impaire.
    
   
    
Ecrire du code qui retourne le n-ième terme de la suite
</div>

<div class="alert alert-warning" role="alert">
Soit n un paramètre, ecrire une boucle qui donne la somme suivante
    $$x_0+x_1+...+x_{n-1}+x_{n}$$ 
</div>

<div class="alert alert-warning" role="alert">
La quantité $4\times (x_1+...+x_n)$ tend vers $\pi$. Soit k un paramètre, ecrire une boucle ```while``` qui approxime la k-ième décimal de $\pi$.
</div>

In [31]:
import math
# pi est donné par :
pi = math.pi
print(pi)

# L'approximation à la k-ième décimal est donnée par :
k=3
pi_approx = round(pi*10**k)/10**k
print(pi_approx)

3.141592653589793
3.142


<div class="alert alert-warning" role="alert">
Par exemple la première décimal de $\pi$ est 3.1 qui est approximé par $x_0+x_1+...+x_{17}$. La fonction doit renvoyer 17.
</div>